tests/unit/s2n_connection_serialize_test.c

```diff
@@ -473,6 +490,271 @@ int main(int argc, char **argv)
         EXPECT_OK(s2n_send_and_recv_test(client_conn, new_server_conn));
     };
 
+    /* Self-talk: Test interaction between resumption and TLS transfer. */
```

Comment:
Does TLS1.2 behave sanely if you try to get a session from a de-serialized connection? Like does it behave like a connection with no session ticket exchanged?
```suggestion

    /* Self-talk: Test interaction between TLS1.3 session resumption and serialization. */
```

Reply:
Added a test for this.